sky/provision/aws/instance.py

```diff
@@ -429,19 +430,75 @@ def _create_node_tag(target_instance, is_head: bool = True) -> str:
             head_instance_id = _create_node_tag(resumed_instances[0])
 
     if to_start_count > 0:
+        target_reservations = (config.node_config.get(
+            'CapacityReservationSpecification',
+            {}).get('CapacityReservationTarget',
+                    {}).get('CapacityReservationId', []))
+        created_instances = []
+        if target_reservations:
+            node_config = copy.deepcopy(config.node_config)
+            node_config['CapacityReservationSpecification'] = {
+                'CapacityReservationTarget': {}
+            }
```

Comment:
Do we need to set this here? We seem to be overwriting this value in the following code.

Reply:
Yes, because we have to clear some of the original settings in `aws-ray.yaml` such as `CapacityReservationPreference` cannot appear when target is set, when actually creating the instance